### PR TITLE
chore(core): add deriveKeypair method to DOT

### PR DESCRIPTION
**Pull request by aaldemir**
When DOT wallet object is used in core for address creation, use hardened key derivation to generate an address to POST to WP for receive addresses. 

Follows workflow implemented by Solana

JIRA ticket: https://bitgoinc.atlassian.net/browse/STLX-13001

### CODE REVIEW

File: modules/core/src/v2/coins/dot.ts [hunk 1]
```diff
@@ -96,6 +102,40 @@ export class Dot extends BaseCoin {
...
+        throw new Error('failed to create valid derived base58 prv key');
+      }
+      if (!_.isString(pub) || !accountLib.Dot.Utils.default.isValidPublicKey(pub)) {
+        throw new Error('failed to create valid derived base58 pub key');
+      }
+      if (!accountLib.Dot.Utils.default.isValidAddress(derivedAddress)) {
+        throw new Error('failed to create valid DOT address from derived base58 keys');
+      }
+      return {
+        prv: '',
```
Comment (alebusse) [inline modules/core/src/v2/coins/dot.ts:130]: I think you gonna need to return this prv since is needed to sign consolidation transfer for example.

Comment (aaldemir) [reply modules/core/src/v2/coins/dot.ts:130]: ah okay - I was wondering where it was getting used. thanks.

File: modules/core/src/v2/wallet.ts [hunk 2]
```diff
@@ -1414,7 +1414,7 @@ export class Wallet {
         });
 
         if (derivedKeypair) {
-          addressParams.derivedAddress = derivedKeypair.pub;
+          addressParams.derivedAddress = derivedKeypair.address ?? derivedKeypair.pub;
```
Comment (alebusse) [inline modules/core/src/v2/wallet.ts:1417]: I created this PR https://github.com/BitGo/BitGoJS/pull/1952, it should help to make the code more coin agnostic

File: modules/core/test/v2/unit/wallet.ts [hunk 3]
```diff
@@ -792,6 +792,141 @@ describe('V2 Wallet:', function () {
...
 
+  describe('Create Address for Polkadot', () => {
+    let dotWallet, coinNocks;
+    const passphrase = 'Passphrase1234';
+    const coinBitgo = new TestBitGo({ env: 'mock' });
+    coinBitgo.initializeTestVars();
+    const tdot = coinBitgo.coin('tdot');
+    const walletData = {
+      id: '598f606cd8fc24710d2ebadb1d9459bb',
+      coinSpecific: {
```
Comment (alebusse) [inline modules/core/test/v2/unit/wallet.ts:803]: Check the coinSpecific data here to see if they are ok

**Review by alebusse — APPROVED** [review]

**Pull request merged**